src/parser/contract-schemas.client.ts

```diff
@@ -0,0 +1,127 @@
+  constructor(nodeUrl: string) {
+    super(nodeUrl);
+  }
+
+  public async fetchContractSchemasBytes(
+    contractHash: string,
+    stateRootHash?: string,
+  ): Promise<Uint8Array> {
+    if (!stateRootHash) {
+      stateRootHash = await this.getStateRootHash();
+    }
+
+    const contractData = (
+      await this.getBlockState(stateRootHash, `hash-${contractHash}`, [])
+    ).Contract;
+
+    if (!contractData) {
+      throw new Error('contract data not found');
+    }
+
+    const eventsSchema = contractData.namedKeys.find(
+      el => el.name === EVENTS_SCHEMA_NAMED_KEY,
+    );
+    if (!eventsSchema) {
+      throw new Error(
+        `'${EVENTS_SCHEMA_NAMED_KEY}' uref not found for contract '${contractHash}'`,
+      );
+    }
+
+    const schemaResponse = await this.client.request({
+      method: 'state_get_item',
+      params: {
+        state_root_hash: stateRootHash,
+        key: eventsSchema.key,
+        path: [],
+      },
+    });
+
+    return decodeBase16(schemaResponse.stored_value.CLValue.bytes);
+  }
+
+  public async getContractsSchemas(
+    contractHashes: string[],
+  ): Promise<Record<string, ContractSchemas>> {
+    const contractsSchemas: Record<string, ContractSchemas> = {};
+
+    for (const contractHash of contractHashes) {
+      const stateRootHash = await this.getStateRootHash();
+
+      const contractData = (
+        await this.getBlockState(stateRootHash, `hash-${contractHash}`, [])
+      ).Contract;
+
+      if (!contractData) {
+        throw new Error('contract data not found');
+      }
+
+      let eventsSchemaUref = '',
+        eventsUref = '';
+      for (const namedKey of contractData.namedKeys) {
+        if (namedKey.name === EVENTS_SCHEMA_NAMED_KEY) {
+          eventsSchemaUref = namedKey.key;
+        } else if (namedKey.name === EVENTS_NAMED_KEY) {
+          eventsUref = namedKey.key;
+        }
+
+        if (eventsSchemaUref !== '' && eventsUref !== '') {
+          break;
+        }
+      }
+
+      if (!eventsSchemaUref) {
+        throw new Error(`no '${EVENTS_SCHEMA_NAMED_KEY}' uref found`);
+      }
+
+      if (!eventsUref) {
+        throw new Error(`no '${EVENTS_NAMED_KEY}' uref found`);
+      }
+
+      const schemaResponse = await this.client.request({
+        method: 'state_get_item',
+        params: {
+          state_root_hash: stateRootHash,
+          key: eventsSchemaUref,
+          path: [],
+        },
+      });
+
+      const schemas = parseSchemasFromBytes(
+        decodeBase16(schemaResponse.stored_value.CLValue.bytes),
+      );
+
+      const contractPackageHash = contractData.contractPackageHash.replace(
+        'contract-package-wasm',
+        '',
+      );
+
+      contractsSchemas[eventsUref] = {
+        schemas,
+        contractHash: decodeBase16(contractHash),
+        contractPackageHash: decodeBase16(contractPackageHash),
+        eventsSchemaUref,
+        eventsUref,
+      };
+    }
+
+    return contractsSchemas;
+  }
```

Comment:
This class seems to be unnecessary. It adds an extra layer of abstraction w/o any clear benefit.

This is a small library and having parser as the facade will simplify the API (taking into account that the logic defined here is used in the `initialize` method).

Reply:
It makes the API more complicated. To initiate the parser you need to create another object and pass it as an argument.

Having parser accept the RPC client from the official SDK provides a much simpler UI.